### DISCUSSION TITLE
AWS Distro layer

### DISCUSSION THREAD

**Discussion by gaspardbalthazar** [awslabs/aws-lambda-web-adapter#512] [General]
Has anyone tried using the aws-lambda-web-adapter layer on a Python and FastAPI Lambda function along with the AWS Distro for OpenTelemetry Lambda layer (https://aws-otel.github.io/docs/getting-started/lambda)? Both layers share the same environment variable AWS_LAMBDA_EXEC_WRAPPER.

In our tests, using a shell script that handles both the AWS Distro execution and the FastAPI project execution with aws-lambda-web-adapter, we observed that OpenTelemetry is executed correctly. However, the FastAPI web adapter doesn't seem to load properly.

We're looking for insights or experiences from anyone who might have successfully integrated these two layers in a Lambda function.

**Comment by bnusunny** [comment]
This could be done. Wrapper scripts always have 'exec "${args[@]}"' as last command to execute the Lambda runtime process (see [the example](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-modify.html#runtime-wrapper) in the Lambda Doc.). So it is possible to chain multiple wrapper scripts together. 

I have not tried this, but here are the steps I would do: 

1. Create your own wrapper script. 

```bash
#/bin/bash
exec [opentelemetry wrapper script] [LWA wrapper script] "${args[@]}"
```

2. Set env var `AWS_LAMBDA_EXEC_WRAPPER=/path/to/your/script`

**Reply by gaspardbalthazar** [reply]
Unfortunately, executing `exec /opt/otel-instrument /opt/bootstrap "${args[@]}"` returns the following error. It seems that the OTel distribution layer expects to find a specific handler:

/opt/bootstrap: line 3: /var/task/otel_wrapper.lambda_handler: No such file or directory

This error indicates that the file or path `/var/task/otel_wrapper.lambda_handler` does not exist, which is likely required by the OpenTelemetry wrapper.
